src/client/datascience/jupyter/liveshare/hostJupyterServer.ts

```diff
@@ -85,31 +91,46 @@ export class HostJupyterServer implements INotebookServer {
         const getExistingSession = async () => {
             const connection = await this.computeLaunchInfo();
 
-            progressDisposable = this.progressReporter.createProgressIndicator(
-                localize.DataScience.connectingToKernel().format(
-                    getDisplayNameOrNameOfKernelConnection(kernelConnection)
-                )
-            );
+            let progressDisposable = ui.disableUI
+                ? undefined
+                : this.progressReporter.createProgressIndicator(
+                      localize.DataScience.connectingToKernel().format(
+                          getDisplayNameOrNameOfKernelConnection(kernelConnection)
+                      )
+                  );
+            cancelToken.onCancellationRequested(() => disposeAllDisposables(disposables), this, disposables);
+            if (progressDisposable) {
+                disposables.push(progressDisposable);
+            } else {
+                ui.onDidChangeDisableUI(
+                    () => {
+                        if (!ui.disableUI && progressDisposable) {
```

Comment:
progressDisposable here is not put in the disposable list. 

Might be better to make the dupe code an anonymous function?